src/components/login/Login.tsx

```diff
@@ -1,70 +1,69 @@
 import React from 'react';
 import { inject, observer } from 'mobx-react';
+import { Redirect } from 'react-router-dom';
+import { Location } from 'history';
+import hslLogo from '~/assets/hsl-logo.png';
+import loginBackground from '~/assets/loginBackground.jpg';
+import AuthService from '~/services/authService';
 import { ErrorStore } from '~/stores/errorStore';
 import { LoginStore } from '~/stores/loginStore';
-import ButtonType from '~/enums/buttonType';
-import Button from '../controls/Button';
 import * as s from './login.scss';
 
 interface ILoginProps {
     errorStore?: ErrorStore;
     loginStore?: LoginStore;
+    location?: Location;
 }
 
 @inject('loginStore', 'errorStore')
 @observer
 class Login extends React.Component<ILoginProps> {
-    // TODO Login logic here
-    public handleUserNameOnChange = (event: React.FormEvent<HTMLInputElement>) => {
-        global.console.log(event.currentTarget.value);
-    }
-
-    // TODO Login logic here
-    public handlePasswordOnChange = (event: React.FormEvent<HTMLInputElement>) => {
-        global.console.log(event.currentTarget.value);
-    }
-
-    public closeLoginModal = () => {
-        this.props.loginStore!.showLogin = false;
-        this.props.errorStore!.addError('Login Modal is not in use');
+    public login = () => {
+        AuthService.authenticate((success: boolean) => {
+            if (success) {
+                this.props.loginStore!.setIsAuthenticated(true);
+            } else {
+                // TODO: this message should be visible to user
+                this.props.errorStore!
+                    .addError('Kirjautuminen epäonnistui. Väärä käyttäjätunnus tai salasana.');
+            }
+        });
     }
 
     public render() {
+        const { from } = this.props.location!.state || { from: { pathname: '/' } };
+        if (this.props.loginStore!.isAuthenticated) {
+            return <Redirect to={from} />;
+        }
+
         return (
         <div className={s.loginView}>
-            <h2>Kirjaudu sisään</h2>
-            <form>
-                <label className={s.label}>
-                    Tunnus
-                <br/>
+            <img className={s.backgroundImage} src={loginBackground} />
```

Comment:
Would suggest using background image instead of adding the image like this. Having background images will allow us to use properties such as 'background-size: contain', which will prevent squeezing the image on screen sizes.

Reply:
Should be fixed now